library/core/src/marker.rs

```diff
@@ -409,12 +410,8 @@ marker_impls! {
 /// [impls]: #implementors
 #[stable(feature = "rust1", since = "1.0.0")]
 #[lang = "copy"]
-// FIXME(matthewjasper) This allows copying a type that doesn't implement
-// `Copy` because of unsatisfied lifetime bounds (copying `A<'_>` when only
-// `A<'static>: Copy` and `A<'_>: Clone`).
-// We have this attribute here for now only because there are quite a few
-// existing specializations on `Copy` that already exist in the standard
-// library, and there's no way to safely have this behavior right now.
+// This is unsound, but required by `hashbrown`
+// FIXME(joboet): change `hashbrown` to use `TrivialClone`
 #[rustc_unsafe_specialization_marker]
```

Comment:
I want to highlight that _this_ PR does not completely remove `Copy` specialization yet, but I think we should consider the FCP as the larger intent to eventually remove it all.

There are also a couple `NonDrop` traits that are piggy-backing on `Copy` and will need to stop or find another means. I think those are sound as-is though, since an actual `Drop` can't be lifetime-dependent.

Reply:
At least the `vec::IntoIter`  `NonDrop` is also a `rustc_unsafe_specialization_marker` which would allow it to continue to exist.

The unstable sort impls also have a specialization relating to Copy and Freeze, I'm not sure what they're doing... I think they make it safe to look at multiple copies of the value instead of moving it?

Reply:
I think we should have a measurement of the full performance cost of removing all `Copy` specialization before making the decision?